src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -109,25 +112,43 @@ private static boolean continueConsuming(final int messagesConsumed, final int m
   }
 
   /**
+   * Write a collection of KeyValueWithTimestamp pairs, with explicitly defined timestamps, to Kafka.
+   *
    * @param topic          Kafka topic to write the data records to
    * @param records        Data records to write to Kafka
    * @param producerConfig Kafka producer configuration
    * @param <K>            Key type of the data records
    * @param <V>            Value type of the data records
    */
-  public static <K, V> void produceKeyValuesSynchronously(
-      final String topic, final Collection<KeyValue<K, V>> records, final Properties producerConfig)
-      throws ExecutionException, InterruptedException {
+  public static <K, V> void produceKeyValuesWithTimestampsSynchronously(
+    final String topic, final Collection<KeyValueWithTimestamp<K, V>> records, final Properties producerConfig)
+    throws ExecutionException, InterruptedException {
     final Producer<K, V> producer = new KafkaProducer<>(producerConfig);
-    for (final KeyValue<K, V> record : records) {
+    for (final KeyValueWithTimestamp<K, V> record : records) {
       final Future<RecordMetadata> f = producer.send(
-          new ProducerRecord<>(topic, record.key, record.value));
+        new ProducerRecord<>(topic, null, record.timestamp, record.key, record.value));
       f.get();
     }
     producer.flush();
     producer.close();
   }
 
+  /**
+   * @param topic          Kafka topic to write the data records to
+   * @param records        Data records to write to Kafka
+   * @param producerConfig Kafka producer configuration
+   * @param <K>            Key type of the data records
+   * @param <V>            Value type of the data records
+   */
+  public static <K, V> void produceKeyValuesSynchronously(
```

Comment:
Re-using makes sense here IMHO because we have a. key/values with timestamps, b. key/values without timestamps, and c. values without timestamps, so we can from generic to specific.


But I agree, the method is hard to read.  I reformatted the lines, it's (IMHO) easier to digest now.